packages/plugin-hardhat/src/utils/impl-address.ts

```diff
@@ -0,0 +1,79 @@
+import { DeploymentNotFound, getBeaconAddress, getImplementationAddress, Manifest } from '@openzeppelin/upgrades-core';
+import { Contract, ethers, Signer, utils } from 'ethers';
+import { EthereumProvider, HardhatRuntimeEnvironment } from 'hardhat/types';
+import { ContractAddressOrInstance, getIBeaconFactory } from '.';
+
+/**
+ * Gets the implementation address from a Beacon.
+ *
+ * @returns the implementation address.
+ */
+export async function getImplementationAddressFromBeacon(
+  hre: HardhatRuntimeEnvironment,
+  signer: ethers.Signer | undefined,
```

Comment:
We shouldn't need a signer for this function. It's an optional argument in `getIBeaconFactory`.